core/trino-main/src/main/java/io/trino/sql/planner/RelationPlanner.java

```diff
@@ -1865,9 +1865,34 @@ private SetOperationPlan process(SetOperation node)
         ImmutableListMultimap.Builder<Symbol, Symbol> symbolMapping = ImmutableListMultimap.builder();
         ImmutableList.Builder<PlanNode> sources = ImmutableList.builder();
 
-        for (Relation child : node.getRelations()) {
+        List<Relation> relations = node.getRelations();
+        checkArgument(relations.size() == 2, "relations size must be 2");
+        Relation rightRelation = relations.getLast();
+        for (Relation child : relations) {
```

Comment:
Do we need this `for` loop as the if statement runs only for right relation

Reply:
The left relation is also required to build symbol mapping.